pkg/scaffold/template.go

```diff
@@ -0,0 +1,76 @@
+package scaffold
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/oslokommune/okctl/pkg/storage"
+
+	kaex "github.com/oslokommune/kaex/pkg/api"
+)
+
+// FetchTemplate downloads an example file and writes it to a buffer
+func FetchTemplate(kx kaex.Kaex) ([]byte, error) {
+	var (
+		err    error
+		buffer bytes.Buffer
+	)
+
+	err = kaex.FetchTemplate(kx, &buffer, "application")
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch example: %w", err)
+	}
+
+	return buffer.Bytes(), nil
+}
+
+// InterpolationOpts defines possible data to inject into the templates
+type InterpolationOpts struct {
+	Domain string
+}
+
+// InterpolateTemplate replaces dummy data in the template with state dependant data
```

Comment:
I'm a bit unsure by reading this doc if this function modifies the passed byte array or not, and what is the return value.